recipes/knowledge_distillation_single_device.py

```diff
@@ -149,17 +149,11 @@ def load_teacher_checkpoint(self, cfg: DictConfig) -> dict[str, Any]:
         """
         Extract the teacher checkpoint state from file.
         """
-        # update checkpointer class to config to work with checkpoint_client
-        # by setting the teacher_checkpointer key as the checkpointer key
-        # that the checkpoint client will use to load
-        new_cfg = OmegaConf.create(cfg)
-        del new_cfg["checkpointer"]
-        teacher_checkpoint_val = new_cfg.pop("teacher_checkpointer")
-        new_cfg["checkpointer"] = teacher_checkpoint_val
-
-        teacher_checkpoint_client = CheckpointClient(
-            new_cfg,
+
+        teacher_checkpointer = config.instantiate(
+            cfg.teacher_checkpointer,
         )
+        teacher_checkpoint_client = CheckpointClient(cfg, teacher_checkpointer)
```

Comment:
IIUC, because we initialize with a checkpointer, when we go to load in a checkpoint, it will hit [this](https://github.com/pytorch/torchtune/blob/3e353259091ccdde439dbcc150812859319c4d81/torchtune/training/checkpointing/_checkpoint_client.py#L93) line and NOT load in a distributed checkpoint with a recipe state.

Is that right?